internal/util/parrot.go

```diff
@@ -0,0 +1,33 @@
+package util
+
+import (
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	maxRetryAttempts int = 3
+)
+
+func RetryFunction(doFunc func(*http.Request) (*http.Response, error), request *http.Request) (*http.Response, error) {
```

Comment:
I suggest naming it something like `ExecuteWithRetry` or something

Reply:
done 👍🏻